src/Sentry/MetricAggregator.cs

```diff
@@ -0,0 +1,459 @@
+                _options.LogDebug("Metric flushed for bucket {0}", key);
+            }
+
+            foreach (var (timestamp, locations) in FlushableLocations())
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                _options.LogDebug("Flushing code locations: ", timestamp);
+                var codeLocations = new CodeLocations(timestamp, locations);
+                _captureCodeLocations(codeLocations);
+                _options.LogDebug("Code locations flushed: ", timestamp);
+            }
+
+            ClearStaleLocations();
+        }
+        catch (OperationCanceledException)
+        {
+            _options.LogInfo("Shutdown token triggered. Exiting metric aggregator.");
+        }
+        catch (Exception exception)
+        {
+            _options.LogError(exception, "Error processing metrics.");
+        }
+        finally
+        {
+            _flushLock.Release();
+        }
+    }
+
+    /// <summary>
+    /// Returns the keys for any buckets that are ready to be flushed (i.e. are for periods before the cutoff)
+    /// </summary>
+    /// <param name="force">Forces all buckets to be flushed, ignoring the cutoff</param>
+    /// <returns>
+    /// An enumerable containing the keys for any buckets that are ready to be flushed
+    /// </returns>
+    internal IEnumerable<long> GetFlushableBuckets(bool force = false)
+    {
+        if (!_buckets.IsValueCreated)
+        {
+            yield break;
+        }
+
+        if (force)
+        {
+            // Return all the buckets in this case
+            foreach (var key in Buckets.Keys)
+            {
+                yield return key;
+            }
+        }
+        else
+        {
+            var cutoff = MetricHelper.GetCutoff();
+            foreach (var key in Buckets.Keys)
+            {
+                var bucketTime = DateTimeOffset.FromUnixTimeSeconds(key);
+                if (bucketTime < cutoff)
+                {
+                    yield return key;
+                }
+            }
+        }
+    }
+
+    private Dictionary<long, Dictionary<MetricResourceIdentifier, SentryStackFrame>> FlushableLocations()
+    {
+        _codeLocationLock.Wait();
+        try
+        {
+            var result = _pendingLocations;
+            _pendingLocations = new Dictionary<long, Dictionary<MetricResourceIdentifier, SentryStackFrame>>();
+            return result;
+        }
+        finally
+        {
+            _codeLocationLock.Release();
+        }
+    }
+
+    /// <summary>
+    /// Clear out stale seen locations once a day
+    /// </summary>
+    private void ClearStaleLocations()
+    {
+        var now = DateTimeOffset.UtcNow;
+        var today = now.GetDayBucketKey();
+        if (lastClearedStaleLocations == today)
+        {
+            return;
+        }
+        // Allow 60 seconds for all code locations to be sent at the transition from one day to the next
```

Comment:
Is this what you mean? Reading the code below it allows for `< 1` so `Minute` has to be `0`. Allowing for the `Seconds` property to go from `0` to `59`

```suggestion
        // Allow less than 60 seconds for all code locations to be sent at the transition from one day to the next
```

Reply:
It checks if the minute is zero (less than 1) and if so it exists. So during the first minute (zero based) of every day, it does NOT try to clear the stale locations... then it'll clear them out.